### PR TITLE
Introduce fallback entry in the ABI

### PR DESCRIPTION
**Pull request by axic**
Partly implements #499 and #562. Comments welcome.

### CODE REVIEW

**Comment by chriseth** [comment]
The fallback function can be "called" even if it is not explicitly defined, so the only information that is conveyed here is whether it is constant or not and that is not even enforced by the compiler yet, so I'm not sure if this is something we want to change the specification over.

**Comment by VoR0220** [comment]
You mean whether or not we want to change the specification for it yet, right? I mean, once constants are enforced, this should be effected, no?

**Comment by axic** [comment]
I think it has two uses:
- users of the JSON ABI do not need to guess or do special processing. All possible methods are listed, including the fallback
- Once the option of having fallback is enforced, this field will be present or missing depending on that. Again, users of the ABI will not need to make any changes.

File: libsolidity/interface/InterfaceHandler.cpp [hunk 1]
```diff
@@ -146,6 +154,15 @@ string InterfaceHandler::ABISolidityInterface(ContractDefinition const& _contrac
 			) +
 			";";
 	}
+	if (_contractDef.fallbackFunction())
```
Comment (chriseth) [inline libsolidity/interface/InterfaceHandler.cpp]: This should be removed now.

**Comment by chriseth** [comment]
Could you please also add the updated interface specification to github.com/ethereum/interfaces ?

**Comment by axic** [comment]
@chriseth yep. I think this can be merged now.

**Pull request merged**